packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/include/video_player_avfoundation/FVPVideoPlayer.h

```diff
@@ -0,0 +1,109 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#if TARGET_OS_OSX
+#import <FlutterMacOS/FlutterMacOS.h>
+#else
+#import <Flutter/Flutter.h>
+#endif
+
+#import <AVFoundation/AVFoundation.h>
+
+#import "FVPDisplayLink.h"
+#import "FVPFrameUpdater.h"
+#import "FVPVideoPlayerPlugin_Test.h"
+
+/// FVPVideoPlayer is responsible for managing video playback using AVPlayer.
+/// It provides methods to control playback, adjust volume, handle seeking, and
+/// notify the Flutter engine about new video frames.
+@interface FVPVideoPlayer ()
+/// The AVPlayerItemVideoOutput associated with this video player.
+@property(readonly, nonatomic, nonnull) AVPlayerItemVideoOutput *videoOutput;
+/// The plugin registrar, to obtain view information from.
+@property(nonatomic, weak, nullable) NSObject<FlutterPluginRegistrar> *registrar;
+/// The CALayer associated with the Flutter view this plugin is associated with, if any.
+@property(nonatomic, readonly, nullable) CALayer *flutterViewLayer;
+/// The Flutter event channel used to communicate with the Flutter engine.
+@property(nonatomic, nonnull) FlutterEventChannel *eventChannel;
+/// The Flutter event sink used to send events to the Flutter engine.
+@property(nonatomic, nonnull) FlutterEventSink eventSink;
+/// The preferred transform for the video. It can be used to handle the rotation of the video.
+@property(nonatomic) CGAffineTransform preferredTransform;
+/// Indicates whether the video player has been disposed.
+@property(nonatomic, readonly) BOOL disposed;
+/// Indicates whether the video player is currently playing.
+@property(nonatomic, readonly) BOOL isPlaying;
+/// Indicates whether the video player is set to loop.
+@property(nonatomic) BOOL isLooping;
+/// Indicates whether the video player has been initialized.
+@property(nonatomic, readonly) BOOL isInitialized;
+/// The updater that drives callbacks to the engine to indicate that a new frame is ready.
+@property(nonatomic, nullable) FVPFrameUpdater *frameUpdater;
+/// The display link that drives frameUpdater.
+@property(nonatomic, nullable) FVPDisplayLink *displayLink;
+/// Whether a new frame needs to be provided to the engine regardless of the current play/pause
+/// state (e.g., after a seek while paused). If YES, the display link should continue to run until
+/// the next frame is successfully provided.
+@property(nonatomic, assign) BOOL waitingForFrame;
+
+NS_ASSUME_NONNULL_BEGIN
```

Comment:
This macro should be used for the entire header, or none of it; we don't use it piecemeal like this anywhere else, so this would be confusing for the team to maintain.